### PR TITLE
Make msid optional for pinned recordings

### PR DESCRIPTION
**Pull request by amCap1712**
As a part of making mbids usable in LB, we need to make msids optional. This PR makes msids optional to pin a recording.

### CODE REVIEW

**Review by mayhem — APPROVED** [review]
Almost there -- very glad to get this done. Feel free to merge after fixing the ALTER TABLE and perhaps giving the modules more descriptive names.

File: admin/sql/updates/2021-12-19-make-msid-optional-for-pinned-recordings.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+BEGIN;
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply admin/sql/updates/2021-12-19-make-msid-optional-for-pinned-recordings.sql:1]: No, pinned recording table has both msid and mbid column already but msid is mandatory. This change makes it optional but mandatory to have at least one.

File: listenbrainz/db/mapping.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
+from typing import Iterable, Dict, List, TypeVar
```
[the remark this answers is not in the document]
Comment (amCap1712) [reply listenbrainz/db/mapping.py:1]: how about `msid_mbid_mapping`?

[the remark this answers is not in the document]
Comment (mayhem) [reply listenbrainz/db/mapping.py:1]: sure!

**Pull request merged**